src/python/pants/core/goals/test.py

```diff
@@ -388,6 +410,45 @@ class Test(Goal):
     __test__ = False
 
 
+@rule_helper
+async def _run_debug_tests(
+    test_subsystem: TestSubsystem,
+) -> Test:
+    targets_to_valid_field_sets = await Get(
+        TargetRootsToFieldSets,
+        TargetRootsToFieldSetsRequest(
+            TestFieldSet,
+            goal_description="`test --debug`",
+            no_applicable_targets_behavior=NoApplicableTargetsBehavior.error,
+        ),
+    )
+    debug_requests = await MultiGet(
+        (
+            Get(TestDebugRequest, TestFieldSet, field_set)
+            if not test_subsystem.debug_adapter
+            else Get(TestDebugAdapterRequest, TestFieldSet, field_set)
+        )
+        for field_set in targets_to_valid_field_sets.field_sets
+    )
+    exit_code = 0
+    for debug_request, field_set in zip(debug_requests, targets_to_valid_field_sets.field_sets):
+        if debug_request.process is None:
+            if test_subsystem.debug_adapter:
+                logger.info(f"Pants doesnt have an adapter for {field_set.address}. Skipping test.")
+            logger.debug(f"Skipping tests for {field_set.address}")
+            continue
+
+        if test_subsystem.debug_adapter:
+            logger.info("Launching debug adapter. Waiting for client connection...")
```

Comment:
```suggestion
        if test_subsystem.debug_adapter:
            logger.info("Launching debug adapter, which will wait for a client connection...")
```

Otherwise there is a race condition where we've said we're waiting, but we might not be yet.